### PR TITLE
[SYCL-MLIR] Support lvalue-returning conditional operator

### PR DESCRIPTION
**Pull request by victor-eds**
Also simplify rvalue and void cases.

### CODE REVIEW

**Review by Naghasan — APPROVED** [review]
we don't care about `throw` expression, so LGTM

**Review by sommerlukas — APPROVED** [review]

**Comment by victor-eds** [comment]
> we don't care about `throw` expression, so LGTM

nice catch, yeah, luckily we do not

**Comment by victor-eds** [comment]
Reinserted old `cgeist` trick to handle lvalues

File: polygeist/tools/cgeist/Test/Verification/conditional.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// RUN: cgeist --use-opaque-pointers %s --function=* -S | FileCheck %s
```
Comment (sommerlukas) [inline polygeist/tools/cgeist/Test/Verification/conditional.cpp:1]: Do we really need the explicit `use-opaque-pointers` flag in this test?

**Review by sommerlukas — APPROVED** [review]
Just one minor nit in the test.

File: polygeist/tools/cgeist/Test/Verification/conditional.cpp (hunk 1 above)
Comment (victor-eds) [reply polygeist/tools/cgeist/Test/Verification/conditional.cpp]: Dropping, as we are not using `llvm.ptr`

**Review by Naghasan — APPROVED** [review]
LGTM

**Review by etiotto — APPROVED** [review]

**Pull request merged**